### ISSUE TITLE
Getting cell metadata from the Notebook

### ISSUE THREAD

**Issue by maveme** [jupyter/help#223]
Hello,
I had implemented a custom kernel for Jupyter, and I want to use the stored metadata information from each cell to do some process in my kernel. 
In the specification of the _The Wire Protocol_, it says that each message has 7 parts and one of those parts is the metadata. However, when I receive the message in my kernel the metadata is always { }, do you know how can I read the metadata from each cell?

Thank you in advance.

**Comment by takluyver** [comment]
The message has metadata, but that's separate from the cell metadata. The cell metadata is not normally sent by our frontends.

Can you describe what you want to do? We might be able to point you to a different way of doing it.

**Comment by maveme** [comment]
Thank you @takluyver for your quick response.

What I want to do is somehow save some information from a previously executed cell, so if I restart my kernel I won't need to execute again other cells, just my actual cell. Do you know how can I implement this?

**Comment by takluyver** [comment]
What sort of information is that? We often restart the kernel to get a 'blank slate', wiping away the results of running code before - in which case it would be annoying if it remembered stuff.

One possibility might be to have some kernel configuration to allow it to always run some code when it starts - e.g. IPython has 'startup files', which it will run every time the kernel is started.

**Comment by maveme** [comment]
Yes, indeed. But what I want to do is having in mind the case in which you shut down the notebook, the next time you open the notebook, you won't need to execute previous cells because you already have all the "required" information to execute the actual cell. 
This information won't be the same for all the notebooks, it will vary from each notebook.

**Comment by takluyver** [comment]
I'm not quite sure what you mean by the 'actual' cell - is there one cell that is special in some way? Is this about making it easier for users by avoiding the need to execute several cells, or is it about avoiding some slow procedure by caching the results? Or something else?

**Comment by maveme** [comment]
By 'actual cell' I mean the cell you want to execute. Exactly, it's more about avoiding the need to run several cells before the current one.
